### PR TITLE
feat: add static constructors for builders with ledger input(account_tx req)

### PR DESCRIPTION
**Pull request by mukulljangid**
The `account_tx` rippled request method lets you specify the ledger(s) that you want the transactions from:
    * By specifying a specific `ledger_index`
    * By specifying a `ledger_index` shortcut like "validated", "current", "closed"
    * By specifying a ledger hash in `ledger_hash`
    * By specifying a lower bound in `ledger_index_min`
    * By specifying an upper bound in `ledger_index_max`
    * By specifying a `ledger range` with both ledger_index_min and ledger_index_max

This PR adds static builder constructors to support these types and make `account_tx` request less confusing.

Adds tests to check new ones and modify previous tests to use new formats. Passes all CI tests. 

Solves Issue #134

### CODE REVIEW

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java [hunk 1]
```diff
@@ -38,10 +38,70 @@
    *
    * @return An {@link ImmutableAccountTransactionsRequestParams.Builder}.
    */
+  @Deprecated
```
Comment (sappenin) [inline xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:41]: Let's add a message that clarifies why this is being deprecated, maybe something like this suggestion.
```suggestion
  @Deprecated Prefer one of the other defined builders so that you are overtly specifying which ledger index or type you want.
```

@nkramer44 Any thoughts here around better wording?

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java [hunk 2]
```diff
@@ -38,10 +38,70 @@
...
+    return ImmutableAccountTransactionsRequestParams.builder().ledgerSpecifier(Optional.ofNullable(ledgerSpecifier));
+  }
+
+  /**
+   * Construct a builder for this class with ledgerHash field set.
+   *
+   * @param ledgerHash Hash of type {@link Hash256} of a specific ledger.
+   * @return An {@link ImmutableAccountTransactionsRequestParams.Builder}.
+   */
+  static ImmutableAccountTransactionsRequestParams.Builder builder(Hash256 ledgerHash) {
```
Comment (sappenin) [inline xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:75]: I don't think we want this builder available here. The contract we want to prod developers to be using is only `LedgerSpecifier`. If someone wants to specify a hash, then they should build a LedgerSpecifier and supply it above above on line 65.

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java [hunk 3]
```diff
@@ -38,10 +38,70 @@
...
 
+  /**
+   * Construct a builder for this class with {@link LedgerIndex} field set.
+   *
+   * @param ledgerIndex An index of type {@link LedgerIndex} to specify a particular ledger.
+   * @return An {@link ImmutableAccountTransactionsRequestParams.Builder}.
+   */
+  static ImmutableAccountTransactionsRequestParams.Builder builder(LedgerIndex ledgerIndex) {
+    return ImmutableAccountTransactionsRequestParams.builder()
+        .ledgerSpecifier(
```
Comment (sappenin) [inline xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:54]: Formatting seems off -- consider pulling line 54 up to line 53.

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java [hunk 4]
```diff
@@ -38,10 +38,70 @@
...
+   * @param ledgerIndexMaximum An upper bound of ledgerIndex of type {@link LedgerIndexBound}.
+   * @return An {@link ImmutableAccountTransactionsRequestParams.Builder}.
+   */
+  static ImmutableAccountTransactionsRequestParams.Builder builder(
+      LedgerIndexBound ledgerIndexMinimum,
+      LedgerIndexBound ledgerIndexMaximum) {
+    LedgerIndexBound ledgerIndexMin = ledgerIndexMinimum;
+    LedgerIndexBound ledgerIndexMax = ledgerIndexMaximum;
+
+    if (ledgerIndexMin == null) {
```
Comment (sappenin) [inline xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:93]: In this library, `null` is a bug, so we should have overt `Objects.requireNonNull(ledgerIndexMinimum);` and `Objects.requireNonNull(ledgerIndexMaximum);` so that null is not allowed.

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java [hunk 5]
```diff
@@ -38,10 +38,70 @@
...
+      ledgerIndexMin = LedgerIndexBound.of(-1);
+    }
+    if (ledgerIndexMax == null) {
+      ledgerIndexMax = LedgerIndexBound.of(-1);
```
Comment (sappenin) [inline xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:97]: @nkramer44 is `-1` an allowed value for `LedgerIndexBound`?

**Review by sappenin — CHANGES_REQUESTED** [review]

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java (hunk 2 above)
Comment (mukulljangid) [reply xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:75]: Do we want to have the similar thing for LedgerIndex? Even that seems to be passed via LedgerSpecifier.

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java (hunk 1 above)
Comment (nkramer44) [reply xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:41]: I think that wording is fine.

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java (hunk 5 above)
Comment (nkramer44) [reply xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:96]: Yes, `LedgerIndexBound` is essentially an unsigned int except for `-1`, because `-1` means "give me the max of this range". For example, if you set the `ledgerIndexMinimum` to `-1`, you are asking for all ledgers that rippled has record of. I believe `-1` is different than not specifying a minimum at all...

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java [hunk 6]
```diff
@@ -38,10 +38,69 @@
...
+   * Construct a builder for this class with {@link LedgerIndex} field set.
+   *
+   * @param ledgerIndex An index of type {@link LedgerIndex} to specify a particular ledger.
+   * @return An {@link ImmutableAccountTransactionsRequestParams.Builder}.
+   */
+  static ImmutableAccountTransactionsRequestParams.Builder builder(LedgerIndex ledgerIndex) {
+    return ImmutableAccountTransactionsRequestParams.builder().ledgerSpecifier(
+        Optional.of(LedgerSpecifier.of(ledgerIndex))
+    );
+  }
```
Comment (nkramer44) [inline xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:56]: We can get rid of this one as well, for the same reason as @sappenin stated below

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java [hunk 7]
```diff
@@ -38,10 +38,69 @@
...
+   */
+  static ImmutableAccountTransactionsRequestParams.Builder builder(LedgerSpecifier ledgerSpecifier) {
+    return ImmutableAccountTransactionsRequestParams.builder().ledgerSpecifier(Optional.ofNullable(ledgerSpecifier));
+  }
+
+  /**
+   * Construct a builder for this class with ledgerHash field set.
+   *
+   * @param ledgerHash Hash of type {@link Hash256} of a specific ledger.
+   * @return An {@link ImmutableAccountTransactionsRequestParams.Builder}.
```
Comment (nkramer44) [inline xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:72]: There should be an empty javadoc line between `@param`s and `@return`. This isn't a hard and fast rule, but we should try to be consistent across the library to reduce PR noise.

You should be able to use IntelliJ's "reformat file" tool (`option + command + L` on mac).

```suggestion
   * @param ledgerHash Hash of type {@link Hash256} of a specific ledger.
   *
   * @return An {@link ImmutableAccountTransactionsRequestParams.Builder}.
```

**Review by nkramer44 — CHANGES_REQUESTED** [review]
Looking good so far. Just a few small things to fix.

Can you also see what happens if you make send an `AccountTransactionsRequestParams` with no `ledgerSpecifier` and no bounds? Does rippled throw an error, or does it return all results from all ledgers?

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountInfoRequestParams.java [hunk 8]
```diff
@@ -44,8 +43,8 @@
    */
   static AccountInfoRequestParams of(Address account) {
     return builder()
-      .account(account)
-      .build();
+        .account(account)
+        .build();
```
Comment (nkramer44) [inline xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountInfoRequestParams.java:47]: Can you reformat this file?

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java [hunk 9]
```diff
@@ -28,20 +29,56 @@
 @Value.Immutable
 @JsonSerialize(as = ImmutableAccountTransactionsRequestParams.class)
 @JsonDeserialize(
-  as = ImmutableAccountTransactionsRequestParams.class,
-  using = AccountTransactionsRequestParamsDeserializer.class
+    as = ImmutableAccountTransactionsRequestParams.class,
+    using = AccountTransactionsRequestParamsDeserializer.class
```
Comment (nkramer44) [inline xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:33]: Same here.

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java [hunk 10]
```diff
@@ -28,20 +29,56 @@
...
+   * Construct a builder for this class with ledgerIndexRange field set.
+   *
+   * @param ledgerIndexMinimum A lower bound of ledgerIndex of type {@link LedgerIndexBound}.
+   * @param ledgerIndexMaximum An upper bound of ledgerIndex of type {@link LedgerIndexBound}.
+   *
+   * @return An {@link ImmutableAccountTransactionsRequestParams.Builder}.
+   */
+  static ImmutableAccountTransactionsRequestParams.Builder builder(
+      LedgerIndexBound ledgerIndexMinimum,
+      LedgerIndexBound ledgerIndexMaximum) {
```
Comment (nkramer44) [inline xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:70]:
```suggestion
  static ImmutableAccountTransactionsRequestParams.Builder builder(
      LedgerIndexBound ledgerIndexMinimum,
      LedgerIndexBound ledgerIndexMaximum
  ) {
```

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java [hunk 11]
```diff
@@ -28,20 +29,56 @@
...
+    LedgerIndexBound ledgerIndexMin = ledgerIndexMinimum;
+    LedgerIndexBound ledgerIndexMax = ledgerIndexMaximum;
```
Comment (nkramer44) [inline xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:72]: Why are these needed? Not sure if I'm missing something...

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java [hunk 12]
```diff
@@ -28,20 +29,56 @@
...
+   * @param ledgerSpecifier A specifier of type {@link LedgerSpecifier} to specify a ledger.
+   *
+   * @return An {@link ImmutableAccountTransactionsRequestParams.Builder}.
+   */
+  static ImmutableAccountTransactionsRequestParams.Builder builder(LedgerSpecifier ledgerSpecifier) {
+    return ImmutableAccountTransactionsRequestParams.builder().ledgerSpecifier(Optional.ofNullable(ledgerSpecifier));
+  }
+
+  /**
+   * Construct a builder for this class with ledgerIndexRange field set.
```
Comment (nkramer44) [inline xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:61]:
```suggestion
   * Construct a builder for this class with {@link LedgerIndexBound}s specified for the {@link #ledgerIndexMinimum} and {@link #ledgerIndexMaximum} fields.
```

**Review by nkramer44 — CHANGES_REQUESTED** [review]

File: xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java (hunk 11 above)
Comment (mukulljangid) [reply xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/accounts/AccountTransactionsRequestParams.java:72]: no, these were for the old == null logic. Taking these off.

**Review by sappenin — APPROVED** [review]

**Review by nkramer44 — APPROVED** [review]
LGTM. Nice job!
